### PR TITLE
[HOTFIX] Point git submodule to branches of forks instead of exact commits

### PR DESCRIPTION
**Pull request by mike-wendt**
- [x] Update docs
- [X] Update git modules
- [x] Update CI to sync git submodules

Needed to fork (cub and moderngpu) and update to tip of the branch to resolve situation where RMM `branch-0.5` was not being updated, leading to errors. Commits are not supported in `.gitmodules` so by forking the repos above we can create a branch that is at our desired commit.

This is to fix an issue where after merging #794 all builds fail because `branch-0.5` for the RMM submodule points to an old commit.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 	path = thirdparty/moderngpu
-	url = https://github.com/moderngpu/moderngpu.git
-	branch = c1fd31df008d79f727483e795b1ee1ce45b782ca
+	url = https://github.com/rapidsai/thirdparty-moderngpu.git
+	branch = cudf-moderngpu
 [submodule "thirdparty/rmm"]
 	path = thirdparty/rmm
 	url = https://github.com/rapidsai/rmm
-	branch = branch-0.5
+	branch = .
```
Comment (kkraus14) [inline .gitmodules]: Why was this changed to `.` instead of `branch-0.5`?

Comment (mike-wendt) [reply .gitmodules]: Because that is shorthand in git to match the current branch name

Comment (mike-wendt) [reply .gitmodules]: So we don't have to change it with every release and dev branch

Comment (kkraus14) [reply .gitmodules]: 👍

Comment (raydouglass) [reply .gitmodules]: https://git-scm.com/docs/git-submodule#git-submodule---branch

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests
